airflow/www/views.py

```diff
@@ -1463,7 +1463,10 @@ def rendered_k8s(self, session: Session = NEW_SESSION):
         map_index = request.args.get('map_index', -1, type=int)
         logging.info("Retrieving rendered templates.")
 
-        dag: DAG = current_app.dag_bag.get_dag(dag_id)
+        dag: DAG = get_airflow_app().dag_bag.get_dag(dag_id)
+        if task_id is None:
+            logging.warning("Task id not passed in the request")
+            abort(400)
```

Comment:
This can be moved much earlier in the function to avoid unnecessary work (getting a DAG is not trivial).

Reply:
Good idea. moved.